capstone-frontend/src/components/SearchBookModal.jsx

```diff
@@ -0,0 +1,236 @@
+
+        let updateJson;
+
+        if (this.props.type === 'club') {
+          updateJson = {
+            id: this.props.objectId,
+            gbookID: bookId,
+          }
+        }
+        else {
+          updateJson = {
+            id: this.props.objectId,
+            add_gbookIDs: bookId,
+          }
+        }
+
+        // Update BookList in Firebase
+        await fetch(this.props.putURL, {
+          method: "PUT",
+          body: JSON.stringify(updateJson)
+        });
+
+      }));
+
+      this.setState({ showModal: false, searchTerm: "", searchResults: [], displayBooks: false, addedBooksIDs: [], addedBooks: [] })
+
+      await this.props.update();
+
+    }
+
+    else {
+
+      this.setState({ showModal: false, searchTerm: "", searchResults: [], displayBooks: false, addedBooksIDs: [], addedBooks: [] })
+    }
+  }
+
+  render() {
+    return (
+      <div>
+        <button className={this.props.btnStyle} onClick={() => this.setState({ showModal: true })}>
+          <div className={this.props.textStyle}>
+            <span id="create-list-modal"> Search for Books </span>
+          </div>
+        </button>
+
+        <Modal
+          size="lg"
+          show={this.state.showModal}
+          onHide={() => this.setState({ showModal: false, searchTerm: "", searchResults: [], displayBooks: false, addedBooksIDs: [], addedBooks: [] })}
+          aria-labelledby="create-booklists-modal">
+
+          <Modal.Header closeButton>
+            <Modal.Title id="create-booklists-modal">
+              Search for Books
+            </Modal.Title>
+          </Modal.Header>
+          <Modal.Body>
+            <Form>
+              <Form.Group controlId="form-search-term">
+                <Form.Control type="text" placeholder="Search" onChange={(event) => this.handleSearchTermChange(event)} />
+                {this.state.fetchingBooks &&
+                  <div className="text-center">
+                    <Spinner
+                      as="span"
+                      animation="border"
+                      size="lg"
+                      role="status"
+                      aria-hidden="true"
+                      className="my-5"
+                    />
+                  </div>}
+              </Form.Group>
+
+              {
+                this.state.displayBooks &&
+
+                <div>
+                  <h3 className="my-4 px-4">Search Results</h3>
+                  <Row className="px-3 text-center">
+                    {this.state.searchResults.map(book =>
+                      <Col md={3} className="px-2 my-0 border" key={book.id}>
+                        <img className="img-responsive mt-3 p-0 rounded" src={book.thumbnailLink} alt={book.title} />
+                        <h5 className="mt-4"> {book.title} </h5>
+                        <p className="my-1"> {book.authors.join(', ')} </p>
+                        {this.state.addedBooksIDs.includes(book.id) ?
```

Comment:
Nice job handling the edge case here if the search result has a book that's already added! 

A related question - are you allowing users to add the same book multiple times to their list? If so, is there a problem if the same gBookID showed up multiple times in a single book list?

Reply:
I think my frontend array allows the addition of multiple duplicate items but I am pretty sure that @steven-solar's generics stops duplicate from being stored.